### PR TITLE
add 508 compliance

### PR DESCRIPTION
**Pull request by carpeliam**
This goes a long way toward making bonnie 508 compliant.

One open question is tab order; there are a few places where buttons appear to the left of the currently focused button (the "gear" button on the measure page, or the "show delete" button in the data criteria section on the patient builder). I'm not sure how much of an issue that might be, and there might be multiple ways to handle that if it is an issue.

### CODE REVIEW

File: app/assets/javascripts/templates/patient_builder/select_criteria.hbs [hunk 1]
```diff
@@ -6,10 +6,11 @@
   </a>
 
   <div id="collapse-{{cid}}" class="collapse">
-    {{#collection}}
+    {{#collection item-view="SelectCriteriaItemView"}}
```
Comment (andrequina) [inline app/assets/javascripts/templates/patient_builder/select_criteria.hbs:9]: what's going on here?

File: app/views/devise/passwords/new.html.erb [hunk 2]
```diff
@@ -7,7 +7,7 @@
   <%= form_for(resource, :as => resource_name, :url => password_path(resource_name), :html => {:name=>"forgot_password_form"}) do |f| %>
     <div class="form-group input-group input-group-lg">
       <span class="input-group-addon"><i class="fa fa-envelope fa-lg fa-fw"></i></span>
-      <%= f.email_field :email, class: "form-control", placeholder: "email@example.com" %>
+      <%= f.email_field :email, class: "form-control", placeholder: "email@example.com", title: "email@example.com" %>
```
Comment (andrequina) [inline app/views/devise/passwords/new.html.erb]: This seems like a bad title... shouldn't this be email?

File: app/views/devise/registrations/edit.html.erb [hunk 3]
```diff
@@ -29,13 +29,13 @@
     <div class="col-md-6">
       <div class="input-group input-group-lg">
         <span class="input-group-addon"><i class="fa fa-envelope fa-lg fa-fw"></i></span>
-        <%= f.email_field :email, class: "form-control", placeholder: "email@example.com" %>
+        <%= f.email_field :email, class: "form-control", placeholder: "email@example.com", title: "email@example.com" %>
```
Comment (andrequina) [inline app/views/devise/registrations/edit.html.erb]: title: email address (required)

File: app/assets/javascripts/templates/patient_builder/select_criteria.hbs (hunk 1 above)
Comment (carpeliam) [inline app/assets/javascripts/templates/patient_builder/select_criteria.hbs:9]: That's part of edc5258. `SelectCriteriaItemView` has a handler for pressing a button (only visible to screen readers) to add the source data criterion to the data criteria.

File: app/views/devise/registrations/new.html.erb [hunk 4]
```diff
@@ -28,13 +28,13 @@
     <div class="col-md-6">
       <div class="input-group input-group-lg required">
         <span class="input-group-addon"><i class="fa fa-envelope fa-lg fa-fw"></i></span>
-        <%= f.email_field :email, class: "form-control", placeholder: "email@example.com" %>
+        <%= f.email_field :email, class: "form-control", placeholder: "email@example.com", title: "email@example.com" %>
```
Comment (andrequina) [inline app/views/devise/registrations/new.html.erb]: again, email

File: app/views/devise/sessions/new.html.erb [hunk 5]
```diff
@@ -9,11 +9,11 @@
   <%= form_for(resource, :as => resource_name, :url => session_path(resource_name), :html => {:name=>"login_form"}) do |f| %>
     <div class="form-group input-group input-group-lg">
       <span class="input-group-addon"><i class="fa fa-envelope fa-lg fa-fw"></i></span>
-      <%= f.email_field :email, class: "form-control", placeholder: "email@example.com" %>
+      <%= f.email_field :email, class: "form-control", placeholder: "email@example.com", title: "email@example.com" %>
```
Comment (andrequina) [inline app/views/devise/sessions/new.html.erb]: again

File: app/views/devise/passwords/new.html.erb (hunk 2 above)
Comment (carpeliam) [inline app/views/devise/passwords/new.html.erb]: I think I checked these with jeff, but I'm ok either way.

**Comment by andrequina** [comment]
I just checked with jeff... he said the title should be "email"

**Comment by carpeliam** [comment]
Sounds good, I'll make the change.

**Pull request merged**